### PR TITLE
[prim_flash] update according to implementation

### PR DESCRIPTION
**Pull request by danaagur**
Signed-off-by: Dana Agur <dana.agur@opentitan.com>

### CODE REVIEW

**Comment by danaagur** [comment]
@tjaychen please review my pr

File: hw/ip/prim/doc/prim_flash.md [hunk 1]
```diff
@@ -1,157 +1,159 @@
...
+clk_i                   | input  | Clock input
+rst_n_i                 | input  | Reset input
+flash_req_i             | input  | Inputs from flash protocol and physical controllers
+flash_rsp_o             | output | Outputs to flash protocol and physical controllers
+prog_type_avail_o       | output | Available program types in this flash wrapper: Currently there are only two types, program normal and program repair
+init_busy_o             | output | The flash wrapper is undergoing initialization
+tck_i                   | input  | jtag tck
+tdi_i                   | input  | jtag tdi
+tms_i                   | input  | jtag tms
+tdo_o                   | output | jtag tdo
```
Comment (tjaychen) [inline hw/ip/prim/doc/prim_flash.md:40]: is there supposed to be a trst_ni here also?
@arnonsha had mentioned it i think.

File: hw/ip/prim/doc/prim_flash.md [hunk 2]
```diff
@@ -1,157 +1,159 @@
...
+scanmode_i              | input  | dft scanmode input
+scan_rst_n_i            | input  | dft scanmode reset
+flash_power_ready_h_i   | input  | flash power is ready (high voltage connection)
+flash_power_down_h_i    | input  | flash wrapper is powering down (high voltage connection)
+flash_test_mode_a_i     | input  | flash test mode values (analog connection)
+flash_test_voltage_h_i  | input  | flash test mode voltage (high voltage connection)
+tl_ul                   | inout  | TL_UL  interface for rd/wr registers access
```
Comment (tjaychen) [inline hw/ip/prim/doc/prim_flash.md:47]: it might be better to split this into `tl_i` and `tl_o` separately.

File: hw/ip/prim/doc/prim_flash.md [hunk 3]
```diff
@@ -1,157 +1,159 @@
...
+  {name: 'pg_erase_i',           wave: '01.0..............'},
+  {name: 'ack_o',                wave: '0.10...10........'},
+  {name: 'erase_suspend_i',      wave: '0.....1.0........'},
+  {name: 'done_o',               wave: '0............10..'},
+ ]
+  }
+{{< /wavejson >}}
+
+## Error Interrupt
+The `flash_err_o` is a level interrupt indication, that is asserted whenever an error event occurs in one of the Flash banks. An Error status register is used to hold the error source of both banks, and it is cleared on read. Clearing the status register trigs deassertion of the interrupt.
```
Comment (tjaychen) [inline hw/ip/prim/doc/prim_flash.md:159]: would you mind clarifying if the `flash_err` signal is pulse or level? This is mostly just for integration purposes.

**Review by tjaychen — APPROVED** [review]
thanks for the update @danaagur !
I've left a few comments.  Let me know what you think and I can merge it.

File: hw/ip/prim/doc/prim_flash.md (hunk 3 above)
Comment (danaagur) [reply hw/ip/prim/doc/prim_flash.md:159]: You mean signal name should be modified? I will change it to become flash_err_lv_o, ok?

File: hw/ip/prim/doc/prim_flash.md (hunk 2 above)
Comment (danaagur) [reply hw/ip/prim/doc/prim_flash.md:47]: sure, you are right

File: hw/ip/prim/doc/prim_flash.md (hunk 1 above)
Comment (danaagur) [reply hw/ip/prim/doc/prim_flash.md:40]: We've checked and there is no need for trst_ni. trst is generated by bist internally from rst_n.

Comment (arnonsha) [reply hw/ip/prim/doc/prim_flash.md:40]: @tjaychen 
My bad, I looked at bist_top which has a trst port, but then again this is not actually the top level :-)
In the real top level the trst_n signal is set to 1 after powerup (inactive), and **doesn't come from an ext pad**

**Comment by tjaychen** [comment]
thanks dana this looks good to me.
@imphil if you are okay with the changes, we can go ahead and merge. 

@danaagur i will make corresponding updates to the open source model.

File: hw/ip/prim/doc/prim_flash.md [hunk 4]
```diff
@@ -141,17 +142,20 @@ A program type not supported by the wrapper, indicated through `prog_type_avail`
...
+  {name: 'erase_suspend_i',      wave: '0.....1.0........'},
   {name: 'done_o',               wave: '0............10..'},
-  {name: 'erase_suspend_done_o', wave: '0............10..'},
-]}
+ ]
+  }
 {{< /wavejson >}}
+
+## Error Interrupt
+The `flash_err_o` is a level interrupt indication, that is asserted whenever an error event occurs in one of the Flash banks. An Error status register is used to hold the error source of both banks, and it is cleared on read. Clearing the status register trigs deassertion of the interrupt.
```
Comment (imphil) [inline hw/ip/prim/doc/prim_flash.md:161]: Nit: Please write documents in a "one sentence per line" format (as documented at https://docs.opentitan.org/doc/rm/markdown_usage_style/ for "Rendered Files"). Also applies to the sentence further up in the file.

**Review by imphil — APPROVED** [review]
Thanks for your update @danaagur! I have a tiny comment inline that would be great to get fixed, but beyond that, LGTM.

**Comment by tjaychen** [comment]
@danaagur there are some minor conflicts now with an update I made, do you mind addressing the conflicts then we can merge this in?

**Comment by danaagur** [comment]
I think it is fixed now, please verify and close if it is ok.
Thanks,
Dana

**Comment by imphil** [comment]
@danaagur Thanks for the rebase, and congratulations to your first merged pull request!

One thing for the next one: Please try to provide a descriptive commit message, containing a bit of information about what changed, and why. This will help us in the future when going back through the changes and try to figure out why a change was done. If you're looking for good examples, commits by @vogelpi or @rswarbrick are generally examples of very nice commit messages (e.g. https://github.com/lowRISC/opentitan/commit/4d1697aa4e2f78d46b099542120b971a4f4c72e8).

Again, thanks for your contribution, and rest assured, working with Git and GitHub will get much easier (and even fun!) rather quickly.

**Pull request merged**